### PR TITLE
Add support for `raw` client-level option

### PR DESCRIPTION
**Pull request by nherson**
This change adds `raw: <boolean>` as a client level option for `Dalli::Client`.

Example usage: `c = Dalli::Client.new(["127.0.0.1:11211"], { raw: true })`

By default, `raw` is `false`. When `true`, the client will keep values "raw" as outlined in [the documentation](https://github.com/petergoldstein/dalli/wiki/Understanding-Serialization-and-Compression#raw-values), unless specified otherwise via an explicit `raw: false` option passed to a `set` op request.

Verifying behavior:

IRB Session
```
irb(main):013> c_raw = Dalli::Client.new(["127.0.0.1:1041"], {raw: true})
=> #<Dalli::Client:0x0000000124d7d168 @key_manager=#<Dalli::KeyManager:0x00000001250d1af8 @key_options={:digest_class=>Digest::MD5}, @namespace=nil>, @normalized_servers=["127.0.0.1:1041"], @options={:raw=>true}, @ring=nil>
irb(main):014> c_raw.set("should:be:raw", Object.new)
=> 2
irb(main):015> c_raw.get("should:be:raw")
=> "#<Object:0x0000000124e9e880>"
irb(main):016> c_serialized = Dalli::Client.new(["127.0.0.1:1041"])
=> #<Dalli::Client:0x0000000124d71fe8 @key_manager=#<Dalli::KeyManager:0x0000000125097a10 @key_options={:digest_class=>Digest::MD5}, @namespace=nil>, @normalized_servers=["127.0.0.1:1041"], @options={}, @ring=nil>
irb(main):017> c_serialized.set("should:not:be:raw", Object.new)
=> 3
irb(main):018> c_serialized.get("should:not:be:raw")
=> #<Object:0x00000001250bd6c0>
```
_(notice the first return value (`raw: true`) is a string with double quotes, the second (`raw: false`) is the original object deserialized)_

Telnet gets of the above data
```
$ telnet localhost 1041
Trying ::1...
Connected to localhost.
Escape character is '^]'.
get should:be:raw
VALUE should:be:raw 0 28
#<Object:0x0000000124e9e880>
END
get should:not:be:raw
VALUE should:not:be:raw 1 12
o:
  Object
END
```

### CODE REVIEW

File: lib/dalli/protocol/value_marshaller.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ class ValueMarshaller
 
       OPTIONS = DEFAULTS.keys.freeze
 
-      def_delegators :@value_serializer, :serializer
+      def_delegators :@value_serializer, :serializer, :raw_by_default?
```
Comment (grcooper) [inline lib/dalli/protocol/value_marshaller.rb:22]: You don't need this method here, delegators are used when this class (the value marshaller) wants to use `@value_serializer.raw_by_default?` but you don't want to always type out `@value_serializer`

File: lib/dalli/protocol/value_serializer.rb [hunk 2]
```diff
@@ -61,6 +62,19 @@ def serialize_value(value)
...
+      def serialize_value?(req_options)
+        # When deciding when serialization is required, check the request-level
+        # option and always refer to that when explicitly set (non-nil), otherwise
+        # use the client default specified by `raw_by_default?`
+        return !raw_by_default? unless req_options && !req_options[:raw].nil?
+
+        !req_options[:raw]
+      end
+
+      def raw_by_default?
```
Comment (grcooper) [inline lib/dalli/protocol/value_serializer.rb:75]: This method can be private.

File: lib/dalli/protocol/value_serializer.rb [hunk 3]
```diff
@@ -61,6 +62,19 @@ def serialize_value(value)
         exc.set_backtrace e.backtrace
         raise exc
       end
+
+      def serialize_value?(req_options)
+        # When deciding when serialization is required, check the request-level
```
Comment (grcooper) [inline lib/dalli/protocol/value_serializer.rb:67]: Can we maybe move this comment above into the params and make it more succinct?

Comment (grcooper) [reply lib/dalli/protocol/value_serializer.rb:67]: This method can also be private.

File: test/protocol/test_value_serializer.rb [hunk 4]
```diff
@@ -24,6 +24,24 @@
         end
       end
     end
+
+    describe 'raw_by_default?' do
```
Comment (grcooper) [inline test/protocol/test_value_serializer.rb:28]: These unit tests aren't super useful. I would just remove them.

File: lib/dalli/protocol/base.rb [hunk 5]
```diff
@@ -16,7 +16,8 @@ class Base
 
       attr_accessor :weight, :options
 
-      def_delegators :@value_marshaller, :serializer, :compressor, :compression_min_size, :compress_by_default?
+      def_delegators :@value_marshaller, :serializer, :compressor, :compression_min_size,
```
[the remark this answers is not in the document]
Comment (nherson) [reply lib/dalli/protocol/base.rb:19]: Rubocop yelled about method length and autocorrected by bumping `:compress_by_default` to the next line.

**Pull request merged**